src/calibre/customize/ui.py

```diff
@@ -691,21 +714,29 @@ def initialize_plugins(perf=False):
         from collections import defaultdict
         import time
         times = defaultdict(lambda:0)
-    for zfp in list(external_plugins) + builtin_plugins:
+
```

Comment:
There needs to be support for BLACKLISTED_PLUGINS for system-wide plugins as well. Also, in add_plugin() you have disabled the ability to ad dplugins whose names conflict with system plugins, however, this code prioritizes installed plugins over system plugins.

Reply:
As the external plugins do not have their name available before being loaded from disk they can't use the same conflict detection logic as the normal plugins who get their name preloaded from `customize.py`.

Would it be fine to change the logic to:
```py
for p in chain(builtin, system, external):
    plugin = _load_plugin(p)
    if already_initialized(plugin) or blacklisted(plugin):
        continue
```

This would have the following effects:
* Even blacklisted plugins are loaded and would be able to execute code.
* A system-plugin that could not be loaded would allow the same plugin in the users installation to be loaded.

It would also need some official decision if it is fine that a system-wide plugin can completely prevent the user from installing the same plugin on their own.
For me personally this fits my usecase perfectly, but I could imagine that it may not be the most correct solution in general.

Reply:
I would prefer blacklisting to work without loading plugin code. To
achieve that simply require the plugin file name in the system plugins
directory be the plugin name. I believe you require that already for the
name conflict check in add_plugin(). 

Coming to policy. Currently I believe external plugins win over builtin
ones in cases where plugins are used for file types/metadata reading
etc. The natural extension of this is

external plugins > system wide plugins > builtin plugins

To achieve that you have to poulatin _initialized_plugins with external
plugins, then system wide and then finally builtin. In addition there
are some functions in customize/ui.py that will need to be adjusted,
such as reread_metadata_plugins() which performs its own sorting on the
plugin list.

Reply:
> I would prefer blacklisting to work without loading plugin code.

Ok.

> To achieve that simply require the plugin file name in the system plugins
directory be the plugin name.

Does `require` mean, that it should be validated in code after the plugin has been loaded, or an assumption can be made that it is like that?

> Coming to policy. Currently I believe external plugins win over builtin
ones in cases where plugins are used for file types/metadata reading
etc.

I think builtin plugins win over external ones:
https://github.com/kovidgoyal/calibre/blob/80364863a9d2f8a8c78222d217c7223204147117/src/calibre/customize/ui.py#L682-L685

https://github.com/kovidgoyal/calibre/blob/80364863a9d2f8a8c78222d217c7223204147117/src/calibre/customize/ui.py#L488-L506